WebHost: Tweaks to search on tracker pages

recklesscoder opened the pull request:
## What is this fixing or adding?
A few quality of life tweaks for the generic tracker pages:
* Pressing `Ctrl+F` or `/` now focuses the search box.
* Typing anywhere now automatically focuses the search box.
* Pressing `Escape` now clears the search and scrolls to the top.

## Points of Deliberation
* Should the standard browser search be suppressed on `Ctrl+F`/`/`?
  * I decided for it, because:
    * I want to use people's muscle memory to show them our search function.
    * I feel like the DataTables search is superior to the browser search in most circumstances.
    * One can still access the standard browser search with `F3` and `Ctrl+G`.
* Should the search filter be restored when the page is loaded?
  * There are two obvious restoration sources: (Note that the search terms obtained from these sources are not necessarily the same.)
    1. Browsers keep textbox contents on `F5` and navigation (but not on `Ctrl+F5` or when opening a fresh tab).
    2. DataTables can load the last search term from local storage (#1145).
  * For now, I have opted to always clear the search on page load with this rationale:
    * Opening a tracker with an active search filter may confuse some users.
    * Refreshing using `F5` is not a use case due to automatic refreshes.

## How was this tested?
Locally, in Firefox and Chrome, with the various search hotkeys as well as Firefox's "Search for text when you start typing" setting.

recklesscoder commented:
> I'm against websites hijacking native browser hot-keys. GitHub does this and it drives me insane. It's bad UX.

Does your criticism extend to all three bulleted features? I'm open to dropping the first one, if that's what people prefer.

LegendaryLinux commented:
Regarding the overriding of browser hotkeys, I agree with Phar. Alternative hotkeys are fine, and I think auto-focusing the search box on page load and when typing is fine. Most users are probably not aware you can still access browser search through alternate means.

Pressing escape to clear the search and return to the top also seems acceptable.

To your second point of deliberation, I think clearing the search box on page reload is the appropriate action.

LegendaryLinux commented on WebHostLib/static/assets/tracker.js:
```diff
@@ -70,10 +71,35 @@ window.addEventListener('load', () => {
+    };
+    searchBox.addEventListener("keyup", doSearch);
+    window.addEventListener("keydown", (event) => {
+        if (event.ctrlKey && !event.altKey && !event.shiftKey && event.key === "f") {
```
Overriding browser hotkeys is inadvisable. Also, since the search box is automatically focused when a user starts typing, this change seems less important. Especially if the search box is focused on page load.

LegendaryLinux reviewed (changes requested):
See also my comment on the PR.

recklesscoder commented:
I added the focusing of the search box on page load and removed the overriding of `Ctrl+F` and `/`. Feel free to merge or let me know what else to change.
